src/JuMP.jl

```diff
@@ -525,6 +529,51 @@ function getDual(c::ConstraintRef{LinearConstraint})
     return c.m.linconstrDuals[c.idx]
 end
 
+function getBndRows(m::Model)
+    numBounds = 0
+    for i in 1:m.numCols
+        seen = false
+        lb, ub = m.colLower[i], m.colUpper[i]
+        for (_,cone) in m.varCones
+            if i in cone
+                seen = true
+                @assert lb == -Inf && ub == Inf
+                break
+            end
+        end
+
+        if !seen
+            if !(lb == -Inf)
+                numBounds += 1
+            end
+            if !(ub == Inf)
+                numBounds += 1
+            end
+        end
+    end
+    return numBounds
+end
+
+function getSOCRows(m::Model)
+    numSOCRows = 0
+    for con in m.socconstr
+        numSOCRows += length(con.normexpr.norm.terms) + 1
+    end
+    return numSOCRows
+end
+
+function getDual(c::ConstraintRef{SOCConstraint})
+    numBndRows = getBndRows(c.m)
+    numSOCRows = getSOCRows(c.m)
```

Comment:
@joehuchette, I'd prefer to avoid saving additional state if we can do so without creating a bottleneck